lib/gitter-adapter.js

```diff
@@ -306,6 +315,36 @@ Adapter.prototype.listUsers = function(channel, key) {
   });
 };
 
+Adapter.prototype.autoJoin = function() {
+  var gc = this.gitterClient;
+
+  gc.currentUser()
+  .then(function(user) {
+    return user.rooms();
+  })
+  .then(function(rooms) {
+    return rooms.map(function(room) {
+      return new GitterRoom(room, gc.client, gc.faye);
+    }).sort(function(a,b){ // sort descending (latest touched first)
+      return Date.parse(a.lastAccessTime) < Date.parse(b.lastAccessTime) ? 1 : -1;
+    });
+  })
+  .then(function(rooms) {
+    var limit = 10;
+    rooms.every(function(room) {
```

Comment:
Use `roomse.slice(0, AUTOJOIN_LIMIT)` and get rid of the tricky decrementing inside the loop.